src/users/users.controller.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpException,
+  HttpStatus,
+  NotFoundException,
+  Param,
+  Patch,
+  Post,
+  Query,
+} from '@nestjs/common';
+import { UsersService } from './users.service';
+import { User } from '@prisma/client';
+
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Get() //   GET /users or /users?min-links=value
+  async getAllUsers(@Query('minLinks') minLinks?: number) {
+    try {
+      return await this.usersService.findAll({ minLinks });
+    } catch (error) {
+      throw new HttpException(
+        'Error fetching users',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  @Get(':id') //  GET /users/:id
+  async getUserById(@Param('id') id: string) {
+    try {
+      return await this.usersService.findUnique({ id });
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw new HttpException(error.message, HttpStatus.NOT_FOUND);
+      } else {
+        throw new HttpException(
+          'Error fetching user',
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+    }
+  }
+
+  @Post() //POST  /users
+  async createUser(@Body() userDto: User) {
```

Comment:
**issue**: Ensure user data is validated before creating a new user.

Reply:
I'll do that in next task